### PR TITLE
Bump protobufs

### PR DESCRIPTION
**Pull request by rubdos**
Update protobuf revision version

    Signal Android 7.10.2
    Signal Desktop 7.17.0-alpha.1

This part is potentially relevant for PNI (cc @direc85)

```diff
diff --git a/libsignal-service/protobuf/SignalService.proto b/libsignal-service/protobuf/SignalService.proto
index 2de3e6ac7..37506c78a 100644
--- a/libsignal-service/protobuf/SignalService.proto
+++ b/libsignal-service/protobuf/SignalService.proto
@@ -439,9 +439,11 @@ message Verified {
 message SyncMessage {
   message Sent {
     message UnidentifiedDeliveryStatus {
-      reserved      /*destinationE164*/ 1;
-      optional string destinationServiceId = 3;
-      optional bool   unidentified         = 2;
+      reserved        /*destinationE164*/      1;
+      optional string destinationServiceId   = 3;
+      optional bool   unidentified           = 2;
+      reserved        /*destinationPni*/       4;
+      optional bytes  destinationIdentityKey = 5;
     }

     message StoryMessageRecipient {
```

### CODE REVIEW

**Comment by gferon** [comment]
oopsie, looks like a bunch of fields disappeared. otherwise, LGTM 🚀

**Review by gferon — APPROVED** [review]

**Comment by rubdos** [comment]
@gferon I added some non-trivial logic here. A pair of eyes and a brain that has been covid free for more than a week would be welcome!

File: libsignal-service/src/sender.rs [hunk 1]
```diff
@@ -518,11 +519,14 @@ where
...
-                        needs_sync,
-                    });
+                    return Ok((
+                        SentMessage {
+                            recipient,
+                            unidentified: unidentified_access.is_some(),
+                            needs_sync,
+                        },
+                        identity_key,
+                    ));
```
Comment (direc85) [inline libsignal-service/src/sender.rs:529]: Would it make sense to add `identity_key` to `SentMessage` instead?

Comment (direc85) [reply libsignal-service/src/sender.rs:529]:
```patch
diff --git a/libsignal-service/src/sender.rs b/libsignal-service/src/sender.rs
index 0e23fc05c..adb02c7c7 100644
--- a/libsignal-service/src/sender.rs
+++ b/libsignal-service/src/sender.rs
@@ -49,14 +49,14 @@ pub struct SendMessageResponse {
     pub needs_sync: bool,
 }
 
-pub type SendMessageResult =
-    Result<(SentMessage, IdentityKey), MessageSenderError>;
+pub type SendMessageResult = Result<SentMessage, MessageSenderError>;
 
 #[derive(Debug, Clone)]
 pub struct SentMessage {
     pub recipient: ServiceAddress,
     pub unidentified: bool,
     pub needs_sync: bool,
+    pub identity_key: IdentityKey,
 }
 
 /// Attachment specification to be used for uploading.
@@ -347,7 +347,11 @@ where
             // if we sent a data message and we have linked devices, we need to send a sync message
             (
                 ContentBody::DataMessage(message),
-                Ok((SentMessage { needs_sync, .. }, _identity_key)),
+                Ok(SentMessage {
+                    needs_sync,
+                    identity_key: _,
+                    ..
+                }),
             ) if *needs_sync || self.is_multi_device().await => {
                 tracing::debug!("sending multi-device sync message");
                 let sync_message = self
@@ -420,7 +424,11 @@ where
                 .await;
 
             match result {
-                Ok((SentMessage { needs_sync, .. }, _)) if needs_sync => {
+                Ok(SentMessage {
+                    needs_sync,
+                    identity_key: _,
+                    ..
+                }) if needs_sync => {
                     needs_sync_in_results = true;
                 },
                 _ => (),
@@ -518,14 +526,12 @@ where
             match send {
                 Ok(SendMessageResponse { needs_sync }) => {
                     tracing::debug!("message sent!");
-                    return Ok((
-                        SentMessage {
-                            recipient,
-                            unidentified: unidentified_access.is_some(),
-                            needs_sync,
-                        },
+                    return Ok(SentMessage {
+                        recipient,
+                        unidentified: unidentified_access.is_some(),
+                        needs_sync,
                         identity_key,
-                    ));
+                    });
                 },
                 Err(ServiceError::Unauthorized)
                     if unidentified_access.is_some() =>
@@ -908,14 +914,14 @@ where
                     let SentMessage {
                         recipient,
                         unidentified,
+                        identity_key,
                         ..
-                    } = sent.0;
-                    let identity_key = sent.1;
+                    } = sent;
                     UnidentifiedDeliveryStatus {
                         destination_service_id: Some(
                             recipient.uuid.to_string(),
                         ),
-                        unidentified: Some(unidentified),
+                        unidentified: Some(*unidentified),
                         destination_identity_key: Some(
                             identity_key.serialize().into(),
                         ),

```

Comment (direc85) [reply libsignal-service/src/sender.rs:529]: With the patch above (on top of the PNI branch), Whisperfish compiles and runs fine (on top of the PNI branch), after a quick test at least.

`UnidentifiedDeliveryStatus` doesn't show up in the PNI branch, but the changes need to be checked in - I'd say that is another PR later.

Comment (rubdos) [reply libsignal-service/src/sender.rs]: That's a very good catch, didn't even occur to me. See a1caf30f5.

**Review by gferon — APPROVED** [review]
LGTM 🚀

File: libsignal-service/src/models.rs [hunk 2]
```diff
@@ -69,7 +68,6 @@ impl Contact {
             color: contact_details.color.clone(),
             verified: contact_details.verified.clone().unwrap_or_default(),
             profile_key: contact_details.profile_key().to_vec(),
-            blocked: contact_details.blocked(),
```
Comment (direc85) [inline libsignal-service/src/models.rs:72]: I find this a little interesting, but it's not up to us :shrug:

**Review by direc85 — APPROVED** [review]
LGTM! I already gave this a spin with the PNI work underneath and this seems to cause no issues at least.

File: libsignal-service/src/models.rs (hunk 2 above)
Comment (rubdos) [reply libsignal-service/src/models.rs:72]: I'd guess there's a new way to communicate blocked users, but haven't look into it. I'll pull it in, thanks!

**Pull request merged**